cmd/main.go

```diff
@@ -5,31 +5,36 @@ import (
 	"log"
 	"os"
 	"sync"
-	"time"
 
 	"github.com/SUMUKHA-PK/Raft-Distributed-Consensus/servermanagement"
 	"github.com/SUMUKHA-PK/Raft-Distributed-Consensus/types"
 )
 
 func main() {
-	file, err := os.Open("server.config.json")
+	raftServers := make(map[string]types.RaftServer)
+	configuration := initializeConfiguration("server.config.json")
+
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
+	go servermanagement.StartServers(configuration)
+	go servermanagement.StartSignal(configuration, raftServers, 1, wg)
```

Comment:
This doesn't work. Try running the code and revert back to old changes.
The timer exists to make the "startSignal" goroutine slower than the "startServers" thread.

Reply:
It works. 

It turns out an unintended side effect of `string.Reader` is that it points to the end of the byte stream. Reusing same reader across different requests meant the subsequent requests were empty.

![2019-12-05-161730_1920x1080_scrot](https://user-images.githubusercontent.com/31231064/70229492-594fb180-177c-11ea-92b8-4881e92646ae.png)